python/libraries/bimvee/geometry.py

```diff
@@ -147,6 +147,35 @@ def combineTwoQuaternions(q1, q2):
     qOut[1:4] = vOut
     return qOut
 
+def quaternionConjugate(q):
+    return np.array([q[0], -q[1], -q[2], -q[3]])
+
+def quaternionInverse(q):
+    return quaternionConjugate(q) / np.sum(q ** 2)
+
+def quaternionProduct(q1, q2):
+    return np.array([ 
+    q1[0] * q2[0] - q1[1] * q2[1] - q1[2] * q2[2] - q1[3] * q2[3],
+    q1[0] * q2[1] + q1[1] * q2[0] + q1[2] * q2[3] - q1[3] * q2[2],
+    q1[0] * q2[2] - q1[1] * q2[3] + q1[2] * q2[0] + q1[3] * q2[1],
+    q1[0] * q2[3] + q1[1] * q2[2] - q1[2] * q2[1] + q1[3] * q2[0] ])
+
+def angleBetweenTwoQuaternions(q1, q2):
```

Comment:
Consider returning the axis of rotation along with the angle, because the angle-axis pair uniquely defines a rotation. I don't know if the angle makes much sense without knowing which axis to rotate by.

Reply:
For my particular use case I only need the angle.